cmd/integration/commands/export_db_data.go

```diff
@@ -0,0 +1,584 @@
+		chainReader := consensuschain.NewReader(chainConfig, tx, blockReader, logger)
+
+		lastFrozenEventBlockNum := blockReader.LastFrozenEventBlockNum()
+		currentBlock, err := blockReader.CurrentBlock(tx)
+		if err != nil {
+			logger.Error(err.Error())
+			return
+		}
+
+		out, err := os.Create(outputCsvFile)
+		if err != nil {
+			logger.Error(err.Error())
+			return
+		}
+
+		csvWriter := csv.NewWriter(out)
+		err = csvWriter.Write([]string{"source", "block_num", "block_time_unix_sec", "block_date_time_utc", "events"})
+		if err != nil {
+			logger.Error(err.Error())
+			return
+		}
+
+		to := min(currentBlock.NumberU64()+1, toNum)
+		for blockNum := fromNum; blockNum < to; blockNum++ {
+			sprintLen := borConfig.CalculateSprintLength(blockNum)
+			if blockNum%sprintLen != 0 {
+				continue
+			}
+
+			header := chainReader.GetHeaderByNumber(blockNum)
+			if header == nil {
+				logger.Error("nil header", "blockNum", blockNum)
+				return
+			}
+
+			events := chainReader.BorEventsByBlock(header.Hash(), blockNum)
+			if len(events) == 0 {
+				continue
+			}
+
+			var source string
+			if blockNum <= lastFrozenEventBlockNum {
+				source = "SNAPSHOTS"
+			} else {
+				source = "DB"
+			}
+
+			var eventsSb strings.Builder
+			for _, eventBytes := range events {
+				var event heimdall.EventRecordWithTime
+				if err := event.UnmarshallBytes(eventBytes); err != nil {
+					logger.Error(err.Error())
+					return
+				}
+
+				eventsSb.WriteString("Event{ID:")
+				eventsSb.WriteString(strconv.FormatUint(event.ID, 10))
+				eventsSb.WriteString(",TimeUnixSec:")
+				eventsSb.WriteString(strconv.FormatInt(event.Time.Unix(), 10))
+				eventsSb.WriteString(",DateTimeUtc:")
+				eventsSb.WriteString(event.Time.UTC().Format(time.RFC3339))
+				eventsSb.WriteString("}")
+				eventsSb.WriteRune(';')
+			}
+
+			record := []string{
+				source,
+				strconv.FormatUint(blockNum, 10),
+				strconv.FormatUint(header.Time, 10),
+				time.Unix(int64(header.Time), 0).UTC().Format(time.RFC3339),
+				eventsSb.String(),
+			}
+			if err = csvWriter.Write(record); err != nil {
+				logger.Error(err.Error())
+				return
+			}
+		}
+
+		csvWriter.Flush()
+		if err = csvWriter.Error(); err != nil {
+			logger.Error(err.Error())
+			return
+		}
+	},
+}
+
+var cmdExportHeimdallSpans = &cobra.Command{
+	Use:   "export_heimdall_spans",
+	Short: "",
+	Run: func(cmd *cobra.Command, args []string) {
+		logger := debug.SetupCobra(cmd, "integration")
+		label := kv.ChainDB
+		db, err := openDB(dbCfg(label, path.Join(datadirCli, label.String())), true, logger)
+		if err != nil {
+			logger.Error(err.Error())
+			return
+		}
+
+		defer db.Close()
+		tx, err := db.BeginRo(cmd.Context())
+		if err != nil {
+			logger.Error(err.Error())
+			return
+		}
+
+		defer tx.Rollback()
+		snapDir := path.Join(datadirCli, "snapshots")
+		logger.Info("snapshot dir info", "dir", snapDir)
+
+		allSnapshots := freezeblocks.NewRoSnapshots(ethconfig.BlocksFreezing{}, snapDir, 0, logger)
```

Comment:
pls use `allSnapshots()` helper in integration. because it using global singletones - and in future they may conflict with your objects (theoreticaly)